cache/blobs.go

```diff
@@ -111,6 +112,9 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 						return zstdWriter(comp)(dest)
 					}
 					mediaType = ocispecs.MediaTypeImageLayer + "+zstd"
+				case compression.Nydus:
+					compressorFunc, finalize = compressNydus(ctx, comp)
+					mediaType = nydusify.MediaTypeNydusBlob
```

Comment:
If we introduce `-tags=nydus`, I think we should move all `if comp.Type == compression.Nydus` and `case compression.Nydus` into `*_nydus.go` file.

Reply:
Now if `-tags=nydus` not to be enabled, the nydus related condition branch will return `errdefs.ErrNotImplemented` in `cache/not_nydus.go`. Seems it's difficult to put `case` and `if` code paths into a separate go file, any suggestions for this?

Reply:
Can we still keep the constant `compression.Nydus` ? If we try to put it into `*_nydus.go`, we need to reimplement some methods, I think the maintainability can become poor.

Reply:
@ktock Adjust some methods for reducing `nydusify` package dependency, PTAL.

If we want to completely remove the `compression.Nydus` dependency when `-tags=nydus` not be enabled, I think we could implement an interface like this for each compression type:

```
type CompressionType interface {
	String() string
	Compress(context.Context, compression.Config) (...)
	Decompress(context.Context, ocispecs.Descriptor) (...)
	NeedsConversion(context.Context, content.Store, ocispecs.Descriptor) (bool, error)
	...
}
```

The advantage of this is that we can ensure that don't miss any `compression.Type` handling in the `switch case` code path, WDYT? If possible, I can submit a separate PR to implement it first.

Reply:
@ktock Introduced a new `compression.Compression` interface in https://github.com/moby/buildkit/pull/3136, PTAL.

Reply:
@AkihiroSuda @ktock Fixed, PTAL, thanks!